dvc/parsing/__init__.py

```diff
@@ -133,17 +134,18 @@ def make_definition(
 
 
 class DataResolver:
-    def __init__(self, repo: "Repo", wdir: str, d: dict):
+    def __init__(self, repo: "Repo", wdir: str = "", d: Dict = None):
         self.fs = fs = repo.fs
-        self.wdir = wdir
+        self.wdir = wdir or os.curdir
         self.relpath = relpath(fs.path.join(self.wdir, "dvc.yaml"))
 
+        d = d or {}
         vars_ = d.get(VARS_KWD, [])
         check_interpolations(vars_, VARS_KWD, self.relpath)
         self.context: Context = Context()
 
         try:
-            args = fs, vars_, wdir  # load from `vars` section
+            args = fs, vars_, self.wdir  # load from `vars` section
```

Comment:
If we fix fs.path.join, will these code changes be necessary? Just checking.

Reply:
I think passing an empty string in place of `os.curdir` is still a correct behaviour to me. This could happen when we use `os.path.dirname` instead which just returns the first element of path which could just be an empty string.

Reply:
Btw, looks like the difference here is that `self.wdir` and `self.relpath` are really symbolic values that are not resolved/normalized for use in `self.fs`(aka they are not `fs_path`s). Right?

Reply:
They are `fs` paths, not a symbolic values. `wdir` is used for finding a complete path to the file from where variables/params are to be loaded.`relpath` is used on error messages, so it's more of a UI thing, but it is still a `fs` path.